### PR TITLE
KAFKA-8617: Use automated protocol for End Txn

### PR DESCRIPTION
**Pull request by abbccdda**
As title

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by abbccdda** [comment]
Retest this please

**Comment by abbccdda** [comment]
We already had a 1/3 pass. Rebasing

**Comment by abbccdda** [comment]
retest this please

**Comment by hachikuji** [comment]
retest this please

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 1]
```diff
@@ -1911,26 +1911,33 @@ class KafkaApis(val requestChannel: RequestChannel,
...
       }
 
-      txnCoordinator.handleEndTransaction(endTxnRequest.transactionalId,
-        endTxnRequest.producerId,
-        endTxnRequest.producerEpoch,
-        endTxnRequest.command,
+      txnCoordinator.handleEndTransaction(endTxnRequest.data.transactionalId,
+        endTxnRequest.data.producerId,
+        endTxnRequest.data.producerEpoch,
+        TransactionResult.forId(endTxnRequest.data.committed),
```
Comment (hachikuji) [inline core/src/main/scala/kafka/server/KafkaApis.scala]: This is a little awkward. I think I'd suggest that we get rid of `forId` and add something like this to `EndTxnResult`:
```java
public TransactionResult result() {
  if (data.committed)
    return TransactionResult.COMMIT;
  else
    return TransactionResult.ABORT;
}
```

**Comment by abbccdda** [comment]
Just record that we have 2/2 green build

**Comment by hachikuji** [comment]
retest this please

**Comment by hachikuji** [comment]
retest this please

**Review by hachikuji — APPROVED** [review]
LGTM

**Pull request merged**